spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java

```diff
@@ -151,36 +153,41 @@ public void destroyAll() {
 	 * Return a page-able list of {@link TaskDefinitionResource} defined tasks.
 	 *
 	 * @param pageable page-able collection of {@code TaskDefinitionResource}
-	 * @param search optional findByTaskNameContains parameter
+	 * @param taskName optional findByTaskNameContains parameter
 	 * @param dslText optional findByDslText parameter
+	 * @param description optional findByDescription parameter
 	 * @param manifest optional manifest flag to indicate whether the latest task execution requires task manifest update
 	 * @param assembler assembler for the {@link TaskDefinition}
 	 * @return a list of task definitions
 	 */
 	@RequestMapping(value = "", method = RequestMethod.GET)
 	@ResponseStatus(HttpStatus.OK)
-	public PagedModel<? extends TaskDefinitionResource> list(Pageable pageable, @RequestParam(required = false) String search,
-			@RequestParam(required = false) boolean manifest, @RequestParam(required = false) String dslText,
-			PagedResourcesAssembler<TaskExecutionAwareTaskDefinition> assembler) {
+	public PagedModel<? extends TaskDefinitionResource> list(Pageable pageable,
+															 @RequestParam(required = false) String taskName,
+															 @RequestParam(required = false) String description,
+															 @RequestParam(required = false) String dslText,
+															 @RequestParam(required = false) boolean manifest,
+															 PagedResourcesAssembler<TaskExecutionAwareTaskDefinition> assembler) {
 
 		final Page<TaskDefinition> taskDefinitions;
-		if (search != null) {
-			if (dslText != null) {
-				throw new TaskQueryParamException(new String[] {"search", "dslText"});
-			} else {
-				taskDefinitions = repository.findByTaskNameContains(search, pageable);
-			}
+
+		if (Stream.of(taskName, description, dslText).filter(Objects::nonNull).count() > 1L) {
+			throw new TaskQueryParamException(new String[]{"taskName", "description", "dslText"});
 		}
-		else {
-			if (dslText != null) {
-				taskDefinitions = repository.findByDslTextContains(dslText, pageable);
+
+		if (taskName == null && dslText == null && description == null) {
+			taskDefinitions = repository.findAll(pageable);
+		} else {
+			if (taskName != null) {
+				taskDefinitions = repository.findByTaskNameContains(taskName, pageable);
+			} else if (description != null) {
+				taskDefinitions = repository.findByDescriptionContains(description, pageable);
 			} else {
-				taskDefinitions = repository.findAll(pageable);
+				taskDefinitions = repository.findByDslTextContains(dslText, pageable);
 			}
 		}
 
-		final Map<String, TaskDefinition> taskDefinitionMap = taskDefinitions.stream()
-				.collect(Collectors.toMap(TaskDefinition::getTaskName, Function.identity()));
+		final Map<String, TaskDefinition> taskDefinitionMap = taskDefinitions.stream().collect(Collectors.toMap(TaskDefinition::getTaskName, Function.identity()));
```

Comment:
**[TinyNit]** Do you mind breaking this back into 2 lines. Did the formatter do this?